controllers/delivery/sync_controller.go

```diff
@@ -223,20 +223,26 @@ func (r *SyncReconciler) Reconcile(ctx context.Context, req ctrl.Request) (resul
 	if obj.Spec.AutomaticPullRequestCreation {
 		log.Info("automatic pull-request creation is enabled, preparing to create a pull request")
 
-		if err = r.Provider.CreatePullRequest(ctx, targetBranch, *obj, *repository); err != nil {
+		id, err := r.Provider.CreatePullRequest(ctx, targetBranch, *obj, *repository)
+		if err != nil {
 			err = fmt.Errorf("failed to create pull request: %w", err)
 			conditions.MarkFalse(obj, meta.ReadyCondition, v1alpha1.CreatePullRequestFailedReason, err.Error())
 
 			return ctrl.Result{}, err
 		}
 
+		obj.Status.PullRequestID = id
+
 		// Create GitRepository to track values file changes.
-		if err := r.createValueFileGitRepository(ctx, *obj, *repository, targetBranch); err != nil {
+		repo, err := r.createValueFileGitRepository(ctx, *obj, *repository, targetBranch)
```

Comment:
Should this be here? Or could it be created by something else...

Reply:
Nothing else knows about the pull request being created and we don't need this unless there _is_ a pull request. Since if there is no pull request, there is no validation.

Reply:
I mean, now that I'm saving the pull request ID... I could also save the branch of the pull request and then the product controller could create it. 🤷